storm-core/src/jvm/backtype/storm/scheduler/Cluster.java

```diff
@@ -90,6 +92,21 @@ public Cluster(INimbus nimbus, Map<String, SupervisorDetails> supervisors, Map<S
         }
         this.conf = storm_conf;
     }
+
+    /**
+     * Get a copy of this cluster object
+     */
+    public Cluster getCopy() {
+        HashMap<String, SchedulerAssignmentImpl> newAssignments = new HashMap<String, SchedulerAssignmentImpl>();
+        for (Map.Entry<String, SchedulerAssignmentImpl> entry : this.assignments.entrySet()) {
+            newAssignments.put(entry.getKey(), new SchedulerAssignmentImpl(entry.getValue().getTopologyId(), entry.getValue().getExecutorToSlot()));
+        }
+        Map newConf = new HashMap<String, Object>();
+        newConf.putAll(this.conf);
+        Cluster copy = new Cluster(this.inimbus, this.supervisors, newAssignments, newConf);
+        copy.status = new HashMap<>(this.status);
+        return copy;
+    }
```

Comment:
Let's make this a static factory method, taking a Cluster and return a copy of it.